packages/react/src/components/Table/TableHead/FilterHeaderRow/FilterHeaderRow.jsx

```diff
@@ -361,6 +362,8 @@ class FilterHeaderRow extends Component {
           const headerContent =
             column.isFilterable !== undefined && !column.isFilterable ? (
               <div />
+            ) : column.customInput !== undefined ? (
+              column.customInput
```

Comment:
If we do change this to be an elementType you can render like so:
![image](https://user-images.githubusercontent.com/1590966/234903451-e18f61de-1ff4-4f21-812b-75860d3bc5f6.png)
Where you define a local variable like `const CustomInput = column.customInput`